src/ts/background/services/keyring-vault.ts

```diff
@@ -24,6 +25,23 @@ class KeyringVault {
     throw new Error(t('keyringNotInit'))
   }
 
+  getTempPassword (): string {
+    return this._tempPassword || ''
+  }
+
+  setTempPassword (tempPassword: string): void {
+    this._tempPassword = tempPassword
+
+    // Trigger clear interval for tempPassword
+    this.clearTempPassword()
+  }
+
+  clearTempPassword (): void {
+    setTimeout(() => {
+      this._tempPassword = undefined
+    }, 300 * 1000) // 5 minutes
```

Comment:
we probably should keep it in a config file

Reply:
Sure, I'll do that 👍